lib/main.dart

```diff
@@ -107,10 +108,42 @@ void registerServicesAndViewModels({
       ),
       dispose: (s) => s.dispose(),
     )
+    ..registerFactory<LastFM>(() {
+      final apiKey = sharedPreferences.getString(kLastFmApiKey) ?? '';
+      final apiSecret = sharedPreferences.getString(klastFmSecret) ?? '';
+      final sessionKey = sharedPreferences.getString(kLastFmSessionKey);
+      final username = sharedPreferences.getString(kLastFmUsername);
+
+      if (sessionKey != null && username != null) {
+        return LastFMAuthorized(
+          apiKey,
+          secret: apiSecret,
+          sessionKey: sessionKey,
+          username: username,
+        );
+      } else {
+        return LastFMUnauthorized(apiKey, apiSecret);
+      }
+    })
     ..registerLazySingleton<ExposeService>(
-      () => ExposeService(
-        discordRPC: allowDiscordRPC ? di<FlutterDiscordRPC>() : null,
-      ),
+      () {
+        final sessionKey = sharedPreferences.getString(kLastFmSessionKey);
+        final lastFMEnabled =
+            sharedPreferences.getBool(kEnableLastFmScrobbling) ?? false;
+        if (sessionKey != null) {
+          return ExposeService(
+            discordRPC: allowDiscordRPC ? di<FlutterDiscordRPC>() : null,
+            lastFm: di<LastFM>() as LastFMAuthorized,
+            lastFmEnabled: lastFMEnabled,
+          );
+        } else {
+          return ExposeService(
+            discordRPC: allowDiscordRPC ? di<FlutterDiscordRPC>() : null,
+            lastFm: null,
```

Comment:
instead of a lastFmEnabled flag, please pass null as lastFm, this should be enough
when you make all calls in the expose service null aware

Reply:
The [commit](https://github.com/ubuntu-flutter-community/musicpod/pull/1000/commits/f51d3daefd981303bb25a6085a8ce1cf62695808) addresses this. Please review.